neqo-common/src/qlog.rs

```diff
@@ -173,11 +175,16 @@ pub fn new_trace(role: Role) -> qlog::TraceSeq {
             group_id: None,
             protocol_type: None,
             reference_time: {
-                // It is better to allow this than deal with a conversion from i64 to f64.
+                // It is better to allow this than deal with a conversion from u128 to f64.
                 // We can't do the obvious two-step conversion with f64::from(i32::try_from(...)),
                 // because that overflows earlier than is ideal.  This should be fine for a while.
                 #[allow(clippy::cast_precision_loss)]
-                Some((time::OffsetDateTime::now_utc().unix_timestamp() * 1000) as f64)
+                Some(
+                    SystemTime::now()
+                        .duration_since(SystemTime::UNIX_EPOCH)
+                        .expect("expect UNIX_EPOCH to always be earlier than now")
+                        .as_millis() as f64,
+                )
```

Comment:
I assume this is a valid assumption. Or, viewed from a different perspective, the best thing Neqo can do on a system running before `SystemTime::UNIX_EPOCH` is panic.

Happy to change to e.g. the below if you prefer.

```suggestion
                    SystemTime::now()
                        .duration_since(SystemTime::UNIX_EPOCH)
                        .map(|d| d.as_millis() as f64),
                        .ok()

```

Reply:
I prefer the alternative. I might prefer that you use [this](https://doc.rust-lang.org/std/time/struct.Duration.html#method.as_secs_f64) instead. `as_millis_f64` would be better, but it isn’t ready [yet](https://github.com/rust-lang/rust/issues/122451).